cmd/session-resume.go

```diff
@@ -0,0 +1,141 @@
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"sort"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/minio/pkg/probe"
+	"github.com/minio/minio/pkg/trie"
+)
+
+var sessionResume = cli.Command{
+	Name:   "resume",
+	Usage:  "Resume a previously saved session.",
+	Action: mainSessionResume,
+	Flags:  globalFlags,
+	Before: setGlobalsFromContext,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [SESSION-ID/all]
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Resume session. 
+	 $ {{.HelpName}} ygVIpSJs
+`,
+}
+
+// bySessionWhen is a type for sorting session metadata by time.
+type bySessionWhen []*sessionV8
+
+func (b bySessionWhen) Len() int           { return len(b) }
+func (b bySessionWhen) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
+func (b bySessionWhen) Less(i, j int) bool { return b[i].Header.When.Before(b[j].Header.When) }
+
+// sessionExecute - run a given session.
+func sessionExecute(s *sessionV8) {
+	switch s.Header.CommandType {
+	case "cp":
+		doCopySession(s)
+	}
+}
+
+// findClosestSessions to match a given string with sessions trie tree.
+func findClosestSessions(session string) []string {
+	sessionsTree := trie.NewTrie() // Allocate a new trie for sessions strings.
+	for _, sid := range getSessionIDs() {
+		sessionsTree.Insert(sid)
+	}
+	var closestSessions []string
+	for _, value := range sessionsTree.PrefixMatch(session) {
+		closestSessions = append(closestSessions, value.(string))
+	}
+	sort.Strings(closestSessions)
+	return closestSessions
+}
+
+// checkSessionResumeSyntax - Validate session resume command.
+func checkSessionResumeSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 2 {
+		cli.ShowCommandHelpAndExit(ctx, "resume", 1) // last argument is exit code
+	}
+}
+
+// mainSessionResume - Main session resume function.
+func mainSessionResume(ctx *cli.Context) error {
+	// Validate session resume syntax.
+	checkSessionResumeSyntax(ctx)
+
+	// Additional command specific theme customization.
+	console.SetColor("Command", color.New(color.FgWhite, color.Bold))
+	console.SetColor("SessionID", color.New(color.FgYellow, color.Bold))
+	console.SetColor("SessionTime", color.New(color.FgGreen))
+
+	if !isSessionDirExists() {
+		fatalIf(createSessionDir().Trace(), "Unable to create session folder.")
+	}
+
+	sessionID := ctx.Args().Get(0)
+	if !isSessionExists(sessionID) {
+		closestSessions := findClosestSessions(sessionID)
+		errorMsg := "Session `" + sessionID + "` not found."
+		if len(closestSessions) > 0 {
+			errorMsg += fmt.Sprintf("\n\nDid you mean?\n")
+			for _, session := range closestSessions {
+				errorMsg += fmt.Sprintf("        `mc resume session %s`", session)
+				// break on the first one, it is good enough.
+				break
+			}
+		}
+		fatalIf(errDummy().Trace(sessionID), errorMsg)
+	}
+	s, err := loadSessionV8(sessionID)
```

Comment:
@awwalker, can you move the remaining code to a new function ? like resumeSession()